iam-login-service/src/main/java/it/infn/mw/iam/audit/events/tokens/AccessTokenIssuedEvent.java

```diff
@@ -15,20 +15,50 @@
  */
 package it.infn.mw.iam.audit.events.tokens;
 
+import java.text.ParseException;
+import java.util.Map;
+
 import org.mitre.oauth2.model.OAuth2AccessTokenEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
+import com.google.common.collect.Maps;
 import com.nimbusds.jose.JWSHeader;
 
+import it.infn.mw.iam.audit.events.IamAuditApplicationEvent;
+import it.infn.mw.iam.audit.events.IamEventCategory;
+
 
-public class AccessTokenIssuedEvent extends TokenEvent {
+public class AccessTokenIssuedEvent extends IamAuditApplicationEvent {
 
   private static final long serialVersionUID = 1L;
+
+  public static final Logger LOG = LoggerFactory.getLogger(AccessTokenIssuedEvent.class);
+
   private final HeaderDTO header = new HeaderDTO();
+  private final Map<String, Object> body;
```

Comment:
Is it really needed to keep a `HeaderDTO` and a `Map` objects, when at the end we will simply print strings? alternatively, can't we simply keep a reference to the token itself and extract what is needed when the event is used?